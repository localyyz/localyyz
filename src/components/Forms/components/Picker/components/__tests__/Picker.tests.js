import React from "react";
import Renderer from "react-test-renderer";

// local
import PickerProvider from "../Provider/index";

// constants
const NAME = "PickerProvider Test - Form";
const PROPS = {
  children: "sample children",
  getField: input => {
    //this is a mock function for the formStore
    //formStore returns value based on input but since we are returning same value no need to use
    return {
      options: {
        test: {
          label: "sample"
        }
      }
    };
  },
  onValueChange: jest.fn()
};

const Component = PickerProvider;

describe(NAME, () => {
  it(`${NAME}: should render properly isVisible:true`, () => {
    let rendered = Renderer.create(<Component.wrappedComponent {...PROPS} />);
    rendered.getInstance().store.isVisible = true; //set it to true
    rendered.getInstance().store.visibleField = "sample"; //set the visible field
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it(`${NAME}: should render properly isVisible:false`, () => {
    let rendered = Renderer.create(<Component.wrappedComponent {...PROPS} />);
    expect(rendered.toJSON()).toMatchSnapshot();
    expect(PROPS.onValueChange).toHaveBeenCalledTimes(0);
  });
});
