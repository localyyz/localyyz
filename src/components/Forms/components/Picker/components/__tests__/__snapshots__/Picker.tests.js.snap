// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerProvider Test - Form PickerProvider Test - Form: should render properly isVisible:false 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    sample children
  </View>
</View>
`;

exports[`PickerProvider Test - Form PickerProvider Test - Form: should render properly isVisible:true 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    sample children
  </View>
  <View
    ease="ease-out"
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.9)",
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      ease="ease-out"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "opacity": 0,
          "paddingHorizontal": 25,
          "paddingVertical": 7.5,
          "transform": Array [
            Object {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <View>
        <RCTPicker
          items={
            Array [
              Object {
                "label": "Not selected",
                "textColor": undefined,
                "value": undefined,
              },
              Object {
                "label": "sample",
                "textColor": undefined,
                "value": "test",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
